File: frontend/src/types/prediction.ts
export interface TopClass {
  class: string;
  full_name?: string;
  code: string;
  confidence: number;
  type: string;
  risk_level: string;
  description?: string;
}

export interface SkinPrediction {
  class: string;
  code: string;
  full_name?: string;
  confidence: number;
  top_classes: TopClass[];
  risk_level: 'low' | 'medium' | 'high';
  requires_consultation: boolean;
  diagnostic_rationale?: string;
  model_version: string;
  model_name: string;
}

export interface HairMetrics {
  coverage_percentage: number;
  follicle_density: number;
  hair_diameter: number;
  scalp_visibility: number;
  sebum_level?: string;
  inflammation_score?: string;
}

export interface HairPrediction {
  density_score: number;
  thinning_stage: string;
  hairline_type: string;
  severity: string;
  recommendation: string;
  diagnostic_rationale?: string;
  metrics: HairMetrics;
  model_version: string;
  model_name: string;
}

export type AssessmentPrediction = SkinPrediction | HairPrediction;

File: backend/app/schemas/prediction.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field
from datetime import datetime

class ClassProbability(BaseModel):
    class_name: str = Field(..., alias="class")
    confidence: float
    description: Optional[str] = None
    risk: Optional[str] = None

    class Config:
        populate_by_name = True

class SkinPredictionResult(BaseModel):
    predicted_class: str = Field(..., alias="class")
    confidence: float
    top_classes: List[Dict[str, Any]]
    risk_level: str
    requires_consultation: bool
    diagnostic_rationale: Optional[str] = None
    model_version: str = "v2.1.0"
    model_name: str = "EfficientNetB3"

    class Config:
        populate_by_name = True

class HairMetrics(BaseModel):
    coverage_percentage: float
    follicle_density: float
    hair_diameter: float
    scalp_visibility: float
    sebum_level: Optional[str] = "Normal"
    inflammation_score: Optional[str] = "Low"

class HairPredictionResult(BaseModel):
    density_score: float
    thinning_stage: str
    hairline_type: str
    severity: str
    recommendation: str
    metrics: HairMetrics
    diagnostic_rationale: Optional[str] = None
    model_version: str = "v2.1.0"
    model_name: str = "MobileNetV3"

class PredictionOut(BaseModel):
    id: str
    assessment_id: str
    model_version: str
    model_name: str
    prediction_type: str
    result: Dict[str, Any]
    confidence: Optional[float] = None
    top_classes: Optional[List[Dict[str, Any]]] = None
    risk_level: Optional[str] = None
    grad_cam_image_url: Optional[str] = None
    processing_time_ms: int
    created_at: datetime
    metadata_json: Optional[Dict[str, Any]] = None

    class Config:
        from_attributes = True
